subsys/spm/spm.c

```diff
@@ -135,8 +135,21 @@ static void spm_config_nsc_flash(void)
 }
 #endif /* CONFIG_ARM_FIRMWARE_HAS_SECURE_ENTRY_FUNCS */
 
+extern int spm_secure_services_init(void);
+
 static void spm_config_flash(void)
 {
+	if (((u32_t)_image_rom_start + (u32_t)_flash_used - 1)
+		>= ((LAST_SECURE_REGION_INDEX + 1)
+			* FLASH_SECURE_ATTRIBUTION_REGION_SIZE)) {
+		PRINT("End of SPM: 0x%x\n",
+			((u32_t)_image_rom_start + (u32_t)_flash_used) - 1);
+		PRINT("Start of app: 0x%x\n", FIRST_NONSECURE_ADDRESS);
+		PRINT("SPM and app are sharing an SPU region. Cannot partition "
+			"flash correctly into secure and non-secure, spinning!\n");
+		while (1)
+			;
```

Comment:
__ASSERT ?

Reply:
`__ASSERT` isn't appropriate because check should always happen.